tempesta_fw/http_parser.c

```diff
@@ -2468,61 +2468,69 @@ __year_day_secs(unsigned int year, unsigned int day_sec)
 	return (days - EPOCH_DAYS) * SEC24H + day_sec;
 }
 
-static size_t
-__skip_weekday(unsigned char *p, size_t len)
-{
-	unsigned char lc, *c, *end = p + len;
-
-	for (c = p; c < end; ++c) {
-		lc = TFW_LC(*c);
-		if ((unsigned)(lc - 'a') > (unsigned)('z' - 'a') && lc != ',')
-			return (*c != ' ') ? CSTR_NEQ : c - p;
-	}
-	return len;
-}
-
 static int
 __parse_http_date(TfwHttpMsg *hm, unsigned char *data, size_t len)
 {
-	static const unsigned long colon_a[] ____cacheline_aligned = {
-		/* ':' (0x3a)(58) Colon */
-		0x0400000000000000UL, 0, 0, 0
-	};
 	int r = CSTR_NEQ;
 	__FSM_DECLARE_VARS(hm);
 
 	__FSM_START_ALT(parser->_i_st);
 
-	__FSM_STATE(I_Date) {
-		/*
-		 * Skip a weekday with comma (e.g. "Sun,") as redundant
-		 * information.
-		 */
-		__fsm_sz = __data_remain(p);
-		__fsm_n = __skip_weekday(p, __fsm_sz);
-		if (__fsm_sz == __fsm_n)
-			__FSM_I_MOVE_n(I_Date, __fsm_sz);
-		if (unlikely(__fsm_n == CSTR_NEQ))
+	/*
+	 * Skip a weekday with comma (e.g. "Sun,") as redundant
+	 * information.
+	 */
+	__FSM_STATE(I_WDate1) {
+		unsigned char lc = TFW_LC(c);
+		if (lc < 'a' || lc > 'z')
 			return CSTR_NEQ;
-		__FSM_I_MOVE_n(I_DateDay, __fsm_n + 1);
+		__FSM_I_MOVE(I_WDate2);
 	}
 
-	__FSM_STATE(I_DateDay) {
-		__fsm_sz = __data_remain(p);
-		/* Parse a 2-digit day. */
-		__fsm_n = parse_int_ws(p, __fsm_sz, &parser->_acc);
-		if (__fsm_n == CSTR_POSTPONE) {
-			parser->_i_st = &&I_DateDay;
-			__msg_hdr_chunk_fixup(data, len);
-		}
-		if (__fsm_n < 0)
-			return __fsm_n;
+	__FSM_STATE(I_WDate2) {
+		unsigned char lc = TFW_LC(c);
```

Comment:
All the three supported formats are _case-sensitive_, so We don't need `TFW_LC` calls at all. The first letter of the day name is upper case, the rest are lowercase.